pkg/v1/remote/image.go

```diff
@@ -283,3 +294,38 @@ func (r *remoteImage) LayerByDigest(h v1.Hash) (partial.CompressedLayer, error)
 		digest: h,
 	}, nil
 }
+
+// This naively matches the first manifest with matching GOOS and GOARCH.
+//
+// We should probably use this instead:
+//	 github.com/containerd/containerd/platforms
+//
+// But first we'd need to migrate to:
+//   github.com/opencontainers/image-spec/specs-go/v1
+func (r *remoteImage) matchImage(rawIndex []byte) ([]byte, *v1.Descriptor, error) {
+	index, err := v1.ParseIndexManifest(bytes.NewReader(rawIndex))
+	if err != nil {
+		return nil, nil, err
+	}
+	for _, childDesc := range index.Manifests {
+		p := v1.Platform{
+			Architecture: "amd64",
+			OS:           "linux",
+		}
+		if childDesc.Platform != nil {
+			p = *childDesc.Platform
+		}
+		if runtime.GOARCH == p.Architecture && runtime.GOOS == p.OS {
```

Comment:
Would it make sense to make runtime.GOARCH and runtime.GOOS parameters here? I've had cases where I want to resolve things for a different platform (think crane digest on a macbook).

Reply:
Yeah we probably want to expose something like `WithPlatform` in our image options, good thinking.

Reply:
Changed this to default to amd64/linux to get the previous behavior (kind of). We can override that with `WithPlatform` wherever it makes sense.